src/charts/time_series_labeler.coffee

```diff
@@ -132,36 +154,105 @@ Ember.Charts.TimeSeriesLabeler = Ember.Mixin.create
       weeks
 
   labelledDays: (start, stop) ->
-    days = d3.time.days(start, stop)
-
-    if days.length > @get 'maxNumberOfLabels'
+    if @numDaysBetween > @get 'maxNumberOfLabels'
       skipVal = Math.ceil(days.length / @get('maxNumberOfLabels'))
       d3.time.days(start, stop).filter((d, i) -> !(i % skipVal))
     else
-      days
+      d3.time.days(start, stop)
 
   labelledHours: (start, stop) ->
-    hours = d3.time.hours(start, stop)
-
-    if hours.length > @get 'maxNumberOfLabels'
+    if @numHoursBetween(start, stop) > @get 'maxNumberOfLabels'
       skipVal = Math.ceil(hours.length / @get('maxNumberOfLabels'))
       d3.time.hours(start, stop).filter((d, i) -> !(i % skipVal))
     else
-      hours
+      d3.time.hours(start, stop)
+
+  # Dynamic x labelling tries to intelligently limit the number of labels 
+  # along the x axis to a specified limit. In order to do this, time 
+  # specificity is callibrated (e.g. for a range of 2 years, instead of having
+  # the labels be in years, the specificity is increased to quarters)
+  dynamicXLabelling: (start,stop) ->
+  	seconds = @numSecondsBetween(start, stop)
+  	hours = @numHoursBetween(start, stop)
+  	days = @numDaysBetween(start, stop)
+  	weeks = days/7
+  	months = d3.time.months(start, stop).length
+  	quarters = months/3
+  	years = d3.time.years(start, stop).length
+
+  	labels = d3.time.years(start, stop)
+  	xDomainType = 'Y'
+  	xLabels = @get 'maxNumberOfLabels'
+
+  	# a ratio specifying the point at which the time specificity should be
+  	# increased
+  	SPECIFICITY_RATIO = 0.25
```

Comment:
So I arbitrarily came up with 0.25. I will play around with this constant to see what value is best.